src/internal/operators/share.ts

```diff
@@ -19,15 +24,15 @@ export interface ShareConfig<T> {
    * or resubscriptions will resubscribe to that same subject. In all cases, RxJS subjects will emit the same error again, however
    * {@link ReplaySubject} will also push its buffered values before pushing the error.
    */
-  resetOnError?: boolean;
+  resetOnError?: boolean | ((error: any) => Observable<any>);
```

Comment:
I'm not sure this makes sense for `resetOnError` and `resetOnComplete`. Semantically, those are "done", so it's not like something were we want to "keep something open" before we check to see if it "unerrors" or "uncompletes". Whereas with refCounting, we're waiting to see if the refCount goes back up from zero.

Reply:
Resets on error/complete are not canceled. This notifier allows conditional/delayed resets (e.g. "cooldown" after error).